src/Microsoft.Diagnostics.Monitoring.EventPipe/Counters/ICounterPayload.cs

```diff
@@ -24,7 +24,7 @@ internal interface ICounterPayload
 
         string DisplayName { get; }
 
-        string Unit { get; }
+        string Unit => CounterMetadata.CounterUnit;
```

Comment:
Are there ever situations where CounterMetadata object is null?

Reply:
Shouldn't be. It is passed in from cache or automatically created for something like `EventCounterPayload`. There are spots in this codebase already where it is dereferenced without a `null` check. The unknown is a lot of code accesses the internals so it is in theory _possible_ something from another repo is passing a `null`. I just added a `Debug.Assert` to try and catch anything.